checks/taskcluster/write_secrets.py

```diff
@@ -0,0 +1,63 @@
+"""
+A check to verify that the Secrets service is operational.
+
+Information about the lastest indexed task is returned.
+"""
+import logging
+from datetime import timedelta
+
+import taskcluster
+import taskcluster.aio
+import taskcluster.exceptions
+
+from poucave import utils
+from poucave.typings import CheckResult
+
+from . import utils as tc_utils
+
+
+logger = logging.getLogger(__name__)
+
+
+DEFAULT_NAME = "project/taskcluster/secrets-test"
+DEFAULT_EXPIRES_SECONDS = 600
+
+
+async def run(
+    root_url: str,
+    secret_name: str = DEFAULT_NAME,
+    expires_seconds: int = DEFAULT_EXPIRES_SECONDS,
+    client_id: str = "",
+    access_token: str = "",
+    certificate: str = "",
+) -> CheckResult:
+    # Build connection infos from parameters.
+    options = tc_utils.options_from_params(
+        root_url, client_id, access_token, certificate
+    )
+
+    secrets = taskcluster.aio.Secrets(options)
+
+    # 1. Write and read.
+    payload = {
+        "expires": (utils.utcnow() + timedelta(seconds=expires_seconds)).isoformat(),
+        "secret": {"hello": "beautiful world"},
+    }
+    await secrets.set(secret_name, payload)
+    try:
+        await secrets.get(secret_name)
+    except taskcluster.exceptions.TaskclusterRestFailure as e:
+        if getattr(e, "status_code") != 404:  # pragma: no-cover
+            raise
+        return False, f"Secret {secret_name!r} was not stored"
+
+    # 2. Remove and check.
+    await secrets.remove(secret_name)
+    try:
+        await secrets.get(secret_name)
+        return False, f"Secret {secret_name!r} was not removed"
+    except taskcluster.exceptions.TaskclusterRestFailure as e:
+        if getattr(e, "status_code") != 404:  # pragma: no-cover
+            raise
```

Comment:
Note, if the secret name never changes, if one time the check fails to remove the secret, all future checks will fail to create it, so an intermittent/temporary issue would appear as a permanent one. However, this would be solved by using a unique secret name for each test with the above suggestion.

Reply:
The `set()` operation at the beginning of the test does not assert anything whether it exists or not, so I think we're good

Reply:
True, but if two checks run at the same time, and only one can create the secret, unless the requests are identical, there is a good chance one of them won't be able to remove the secret and cause a failure. In the case the request URL and request body (for a secrets.remove there probably is no http body) being identical, normally the idempotent nature of the API design would mean that both requests should succeed, however since they may come from different source IP addresses, I'm not sure whether the idempotency checks in the service will consider the requests to be the same or not, and therefore could still cause a failure.

In general I suspect it will be easier to untangle failures from each other if each secret publish/remove test uses a unique secret name.